### PR TITLE
[SER-419] Ball contract cs

### PR DESCRIPTION
**Pull request by TheButlah**
~rebased on #101~

### CODE REVIEW

File: demos/unity_states/cs/generated/cpp_sharp/Std.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// ----------------------------------------------------------------------------
```
Comment (tomthecarrot) [inline demos/unity_states/cs/generated/cpp_sharp/Std.cs]: Let's `.gitignore` the `generated` directory.

File: client/codegen_pinvoke/Main.cs [hunk 2]
```diff
@@ -7,55 +7,77 @@
...
+            this.input_dir = new DirectoryInfo(input_dir);
+            this.crate_name = crate_name;
+        }
+    }
+
     public class Codegen : CppSharp.ILibrary
     {
         static DirectoryInfo project_dir = GetProjectDir();
-        DirectoryInfo output_dir;
+        LibInfo lib_info;
```
Comment (tomthecarrot) [inline client/codegen_pinvoke/Main.cs]: Mark `private`

**Review by tomthecarrot — CHANGES_REQUESTED** [review]

File: demos/unity_states/cs/generated/cpp_sharp/Std.cs (hunk 1 above)
Comment (TheButlah) [reply demos/unity_states/cs/generated/cpp_sharp/Std.cs]: good catch

File: client/codegen_pinvoke/Main.cs (hunk 2 above)
Comment (TheButlah) [reply client/codegen_pinvoke/Main.cs]: sure - keep in mind though that this is an application not a library

Comment (tomthecarrot) [reply client/codegen_pinvoke/Main.cs]: 👍 just best practice in C# to mark private & use getter/setter if needed outside of the class.

File: demos/unity_states/cs/generated/cpp_sharp/Std.cs (hunk 1 above)
Comment (TheButlah) [reply demos/unity_states/cs/generated/cpp_sharp/Std.cs]: done

File: client/codegen_pinvoke/Main.cs (hunk 2 above)
Comment (TheButlah) [reply client/codegen_pinvoke/Main.cs]: done

**Review by tomthecarrot — APPROVED** [review]

**Pull request merged**